components/brave_wallet/browser/swap_service.cc

```diff
@@ -383,42 +383,53 @@ void SwapService::GetQuote(mojom::SwapQuoteParamsPtr params,
                            GetQuoteCallback callback) {
   auto conversion_callback = base::BindOnce(&ConvertAllNumbersToString);
 
-  if (params->from_chain_id == params->to_chain_id &&
-      IsNetworkSupportedByZeroEx(params->from_chain_id)) {
-    auto swap_fee = GetZeroExSwapFee();
+  auto has_zero_ex_support = params->from_chain_id == params->to_chain_id &&
+                             IsNetworkSupportedByZeroEx(params->from_chain_id);
+  auto has_jupiter_support = params->from_chain_id == params->to_chain_id &&
+                             IsNetworkSupportedByJupiter(params->from_chain_id);
+  auto has_lifi_support = IsNetworkSupportedByLiFi(params->from_chain_id) &&
+                          IsNetworkSupportedByLiFi(params->to_chain_id);
+
+  // If the provider is set to Auto, Solana swaps are served via Jupiter.
+  if ((params->provider == mojom::SwapProvider::kJupiter ||
+       params->provider == mojom::SwapProvider::kAuto) &&
+      has_jupiter_support) {
+    auto swap_fee = GetJupiterSwapFee(*params);
     auto fee_param = swap_fee->fee_param;
 
     auto internal_callback = base::BindOnce(
-        &SwapService::OnGetZeroExQuote, weak_ptr_factory_.GetWeakPtr(),
+        &SwapService::OnGetJupiterQuote, weak_ptr_factory_.GetWeakPtr(),
         std::move(swap_fee), std::move(callback));
 
     api_request_helper_.Request(net::HttpRequestHeaders::kGetMethod,
-                                GetZeroExQuoteURL(*params, fee_param), "", "",
+                                GetJupiterQuoteURL(*params, fee_param), "", "",
                                 std::move(internal_callback), GetHeaders(), {},
                                 std::move(conversion_callback));
 
     return;
   }
 
-  if (params->from_chain_id == params->to_chain_id &&
-      IsNetworkSupportedByJupiter(params->from_chain_id)) {
-    auto swap_fee = GetJupiterSwapFee(*params);
+  // EVM swaps are served via 0x only if the provider is set to kZeroEx.
+  if (params->provider == mojom::SwapProvider::kZeroEx && has_zero_ex_support) {
+    auto swap_fee = GetZeroExSwapFee();
     auto fee_param = swap_fee->fee_param;
 
     auto internal_callback = base::BindOnce(
-        &SwapService::OnGetJupiterQuote, weak_ptr_factory_.GetWeakPtr(),
+        &SwapService::OnGetZeroExQuote, weak_ptr_factory_.GetWeakPtr(),
         std::move(swap_fee), std::move(callback));
 
     api_request_helper_.Request(net::HttpRequestHeaders::kGetMethod,
-                                GetJupiterQuoteURL(*params, fee_param), "", "",
+                                GetZeroExQuoteURL(*params, fee_param), "", "",
                                 std::move(internal_callback), GetHeaders(), {},
                                 std::move(conversion_callback));
 
     return;
   }
 
-  if (IsNetworkSupportedByLiFi(params->from_chain_id) &&
-      IsNetworkSupportedByLiFi(params->to_chain_id)) {
+  // EVM swaps are served via LiFi if the provider is set to kLiFi or kAuto.
```

Comment:
Could you please reorder these back to `ZeroEx, Jupiter, LiFi`? Would be a bit easier to review

Reply:
Done